### ISSUE TITLE
Library unusable in Next.js with TypeScript

### ISSUE THREAD

**Issue by MathisBarre** [electerious/use-ackee#8]
Hey ! I'm trying to use use-ackee in my Next.js with TypeScript application but useAckee always throw an Error saying that's something is missing.

When i put the minimum to work, "location is not defined", when I add option (all in false) , "navigator is not defined"

It's maybe just a TypeScript issue because there in no types available (@types/use-ackee is only avalaible in v2 and v2 documentation for use-ackee seems to not being available)

Here is the minimal code from  _App.tsx :
```tsx
import { useEffect } from 'react'
import { AppProps } from 'next/app'
import { useRouter } from 'next/router'
import Head from "next/head"
import "tailwindcss/tailwind.css"
import '../styles/globals.css'
// @ts-ignore
import useAckee from "use-ackee"

function App({ Component, pageProps }: AppProps) {
  const router = useRouter()

  useAckee("/", {
    server: 'https://mathisbarre.com',
    domainId: '3ab5bsd75-a533-4263-a4cdc-8e6c0cdzdadcc29c'
  })

  return (
    <>
      <Head>
        <meta property="og:image" content="https://mathisbarre.com/images/overview.png" />
        <meta property="twitter:image" content="https://mathisbarre.com/images/overview.png" />
      </Head>
      <Component {...pageProps} />
    </>
  )
}

export default App
```

Note: I need to use ts-ignore because of the lack of types definition

Here is the complete error :

```
ReferenceError: location is not defined
    at Object.S [as create] (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\ackee-tracker\dist\ackee-tracker.min.js:1:18787)
    at C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\use-ackee\src\index.js:17:23
    at Object.useMemo (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react-dom\cjs\react-dom-server.node.development.js:1582:19)
    at useMemo (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react\cjs\react.development.js:1532:21)     
    at useAckee (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\use-ackee\src\index.js:15:19)
    at App (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\.next\server\pages\_app.js:49:51)
    at processChild (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react-dom\cjs\react-dom-server.node.development.js:3353:14)
    at resolve (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react-dom\cjs\react-dom-server.node.development.js:3270:5)
    at ReactDOMServerRenderer.render (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react-dom\cjs\react-dom-server.node.development.js:3753:22)
    at ReactDOMServerRenderer.read (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react-dom\cjs\react-dom-server.node.development.js:3690:29)
    at renderToString (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\react-dom\cjs\react-dom-server.node.development.js:4298:27)
    at Object.renderPage (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\next\dist\next-server\server\render.js:54:854)
    at Function.getInitialProps (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\.next\server\pages\_document.js:518:19) 
    at Function.getInitialProps (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\.next\server\pages\_document.js:1113:85)    at loadGetInitialProps (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\next\dist\next-server\lib\utils.js:5:101)
    at renderToHTML (C:\Users\mathis\Documents\DevWeb\Projets\Projets_perso\nextjs.mathisbarre.com\node_modules\next\dist\next-server\server\render.js:54:1145)
```

Hope I have been clear with my issue ! Don't hesitate to ask questions !

**Comment by electerious** [comment]
Hey @MathisBarre, it's definitely a TS issue, but I don't know how to fix it nor can I provide any help for TS related issues. I hope you find a solution! If so, please let me know.

**Comment by yassinebridi** [comment]
Have you tried lazy load importing use-ackee?
https://nextjs.org/docs/advanced-features/dynamic-import

**Comment by merrywhether** [comment]
It's not a TS issue, but rather that `useAckee` isn't SSR safe. I've made #9 to fix that, but in the meantime you can just do the following and disable the eslint error for conditional hook usage (safe since the execution context of your app can't change while it's running).

```ts
export const App: () => {
  const { asPath } = useRouter();
  
  if (typeof window !== 'undefined') {
    useAckee(
      asPath,
      {
        domainId: '...',
        server: '...',
      },
    );
  }
  ...
}
```

**Comment by electerious** [comment]
@merrywhether Thanks for letting me know! It's fixed in version 3.0.1.
